pkg/policy/k8s/service.go

```diff
@@ -311,12 +294,24 @@ func (s *serviceEndpoints) processRule(rule *api.Rule) (numMatches int) {
 		for _, toService := range egress.ToServices {
 			if sel := toService.K8sServiceSelector; sel != nil {
 				if serviceSelectorMatches(sel, s.svcID, s.svc) {
-					appendEndpoints(&rule.Egress[i].ToCIDRSet, s.endpoints())
+					if s.svc.IsExternal() {
+						appendEndpoints(&rule.Egress[i].ToCIDRSet, s.endpoints())
+					} else {
+						endpointSelector := api.NewESFromMatchRequirements(s.svc.Selector, nil)
```

Comment:
I'm also pretty worried about plumbing this directly without sanitizing it correctly. I think that can result in potentially undesired behavior with certain labels. It could have been sanitized earlier, but we have to be 100% sure that is the case, and that edge cases are properly handled, before adding it like this.


I guess one use-case here is;
- Admin creates CNP in namespace A to select all services in namespace
- User has access to create Kubernetes services in namespace A, but _not_ endpoints/endpointslices or network polices of any kind.

In this case I believe the user should only be able to open egress/ingress to/from pods in namespace A, and not other namespaces or other endpoints in the cluster.

cc @joestringer

Reply:
Very good question; it seems like @chaunceyjiang has since changed the code to force-add a namespace selector, which mimics what the services controller would do.

So, yes, good catch!